milli/src/update/settings.rs

```diff
@@ -1033,6 +1004,13 @@ impl<'a, 't, 'i> Settings<'a, 't, 'i> {
             }
             Setting::NotSet => false,
         };
+
+        // if any changes force a reindexing
+        // clear the vector database.
+        if update {
+            self.index.vector_arroy.clear(self.wtxn)?;
+        }
```

Comment:
it should be removed in the future in favor of a more granular embeddings indexing process